File: frontend-Application/src/services/admin/examService.js
import adminAxios from "../axios/adminAxios";

export const getAllExams = async () => {
    return await adminAxios.get("/admin/exams");
};

export const getExamById = async (id) => {
    return await adminAxios.get(`/exams/${id}`);
};

export const createExam = async (exam) => {
    return await adminAxios.post("/admin/exams", exam);
};

export const deleteExam = async (id) => {
    return await adminAxios.delete(`/admin/exams/${id}`);
};

File: oep_spring_backend/src/main/java/com/oep/service/ExamService.java
package com.oep.service;

import java.util.List;
import com.oep.entities.Exam;
import com.oep.entities.ExamQuestions;

public interface ExamService {
    List<Exam> getAllExams();

    List<Exam> getExamsByInstructor(Long instructorId);

    List<Exam> getExamsByStudent(Long studentId);

    Exam getExamById(Long id);

    Exam createExam(Exam exam);

    Exam updateExam(Long id, Exam exam);

    void deleteExam(Long id);

    void addQuestionToExam(Long examId, Long questionId);
    
    void removeQuestionFromExam(Long examId, Long questionId);

    boolean verifyPassword(Long examId, String password);

    com.oep.entities.ExamResults submitExam(Long examId, Long studentId, java.util.Map<Long, String> answers);

	List<ExamQuestions> getExamQuestions(Long examId);
}
